tensorflow/contrib/eager/python/examples/nmt_with_attention/nmt_with_attention.ipynb

```diff
@@ -677,21 +678,23 @@
         "                # using teacher forcing\n",
         "                dec_input = tf.expand_dims(targ[:, t], 1)\n",
         "        \n",
-        "        total_loss += (loss / int(targ.shape[1]))\n",
-        "        \n",
         "        variables = encoder.variables + decoder.variables\n",
         "        \n",
         "        gradients = tape.gradient(loss, variables)\n",
-        "      \n",
+        "        \n",
         "        optimizer.apply_gradients(zip(gradients, variables), tf.train.get_or_create_global_step())\n",
-        "\n",
+        "        \n",
+        "        batch_loss += (loss / int(targ.shape[1]))\n",
+        "        \n",
+        "        total_loss += batch_loss\n",
+        "        \n",
         "        if batch % 100 == 0:\n",
         "            print('Epoch {} Batch {} Loss {:.4f}'.format(epoch + 1,\n",
         "                                                         batch,\n",
-        "                                                         loss.numpy() / int(targ.shape[1])))\n",
+        "                                                         batch_loss.numpy()))\n",
```

Comment:
Same here. I think this should remain the same.

Reply:
Ya `batch_loss` is the same thing as `loss/int(targ.shape[1])`.